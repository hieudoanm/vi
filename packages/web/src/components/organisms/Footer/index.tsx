import { Container } from '@mui/material';
import React from 'react';

export type Page = {
  category: string;
  links: string[];
};

export type FooterProps = { pages: Page[] };

const Footer: React.FC<FooterProps> = ({ pages = [] }) => {
  return (
    <footer className="bg-white py-20">
      <Container>
        <div className="grid grid-cols-4 gap-8">
          {pages.map(({ category, links }) => {
            return (
              <div key={`category-${category}`}>
                <span className="font-bold">{category}</span>
                <ul className="pt-4">
                  {links.map((link: string, index: number) => {
                    return (
                      <li key={`link-${index}`} className="mb-3">
                        {link}
                      </li>
                    );
                  })}
                </ul>
              </div>
            );
          })}
        </div>
      </Container>
      <Container className="pt-12">
        <div className="font-extrabold tracking-widest">VI</div>
      </Container>
    </footer>
  );
};

export default Footer;
