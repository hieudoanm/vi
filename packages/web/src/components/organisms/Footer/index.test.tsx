import { render } from '@testing-library/react';
import Footer from '.';

describe('Footer', () => {
  test('render default', () => {
    const wrapper = render(<Footer pages={[]} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
