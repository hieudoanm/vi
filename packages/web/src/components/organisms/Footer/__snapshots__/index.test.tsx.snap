// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render default 1`] = `
<div>
  <footer
    class="bg-white py-20"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="grid grid-cols-4 gap-8"
      />
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg pt-12 css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="font-extrabold tracking-widest"
      >
        VI
      </div>
    </div>
  </footer>
</div>
`;
